infra/growth/src/resources/rules.js

```diff
@@ -430,11 +470,53 @@ class BaseRule {
         : GrowthActionStatus.Completed
     } else {
       // Evaluate the rule to determine if it is Completed.
-      return (await this.evaluate(ethAddress, events))
+      return (await this._evaluate(ethAddress, events))
         ? GrowthActionStatus.Completed
         : GrowthActionStatus.Active
     }
   }
+
+  conditionsToUnlock() {
+    const prevLevel = this.levelId - 1
```

Comment:
I personally prefer the functional version if this function, since it has no variable assignments and states but I guess this works too.
```
return allRules
    .filter(allRule => allRule.levelId === rule.levelId - 1)
    .filter(allRule => allRule.config.nextLevelCondition === true)
    .flatMap(allRule =>
      allRule.config.unlockConditionMsg.map(conditionMessage => {
        return {
          messageKey: conditionMessage.conditionTranslateKey,
          iconSource: conditionMessage.conditionIcon
        }
      })
    )
```